server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -2425,9 +2425,6 @@ public void importMapFiles(long tid, Map<FileRef,MapFileInfo> fileMap, boolean s
       }
     } finally {
       synchronized (this) {
-        if (writesInProgress < 1)
```

Comment:
I think these checks should be left as-is. It may not be possible to for the current code result in this condition, but changes in the code could cause the assumptions to be violated.  If that happens it would be nice to have the checks.

Reply:
I can add back the ones below but I think this one is actually problematic.  There is already a similar check above, right before the increment.  The only way this case can happen is if there is an error thrown and if that is the case we will swallow the original error here.

I also worry about the exceptions thrown by the other checks swallowing exceptions.

Reply:
Nevermind, the ones below aren't in a finally just unreachable.  I don't think we should keep around unreachable code just in case someone changes code.

Reply:
> Nevermind, the ones below aren't in a finally just unreachable. I don't think we should keep around unreachable code just in case someone changes code.

I agree those could be removed since they are local vars and the condition can not happen.

> There is already a similar check above

The previous check was done in a sync block, then the thread exited the sync block.  The code re-enters the sync block so it makes sense to do the check again.  I think the check is still useful.

> I also worry about the exceptions thrown by the other checks swallowing exceptions.

Its not ever expected to happen unless there is a bug in the code elsewhere.

Reply:
OK I see what is going on now.  I will reword the message since its pretty important.

Looks like a bug with the line of code OUTSIDE the try though so I can at least fix that.

Reply:
On second thought, I will do these Tablet improvements separately.